ec2/spark_ec2.py

```diff
@@ -569,15 +569,34 @@ def launch_cluster(conn, opts, cluster_name):
         master_nodes = master_res.instances
         print "Launched master in %s, regid = %s" % (zone, master_res.id)
 
```

Comment:
I agree with you. Since we can meet the same problem about propagating information of AWS in the `wait_for_cluster_state` function, I think that adding waiting time is much better workaround. 
The only disadvantage is that we need more 10 second to launch a cluster.